tests/unit/deploy/test_deployer.py

```diff
@@ -854,6 +855,82 @@ def test_can_update_auth_handlers(self, sample_app_with_auth):
             zip_contents=b'package contents',
         )
 
+    def test_can_create_auth_with_different_config(self, sample_app_with_auth):
+        # We're notusing create_config_obj because we want to approximate
+        # loading config from disk which contains per-lambda configuration.
+        disk_config = {
+            'app_name': 'myapp',
+            'iam_role_arn': 'role-arn',
+            'manage_iam_role': False,
+            'stages': {
+                'dev': {
+                    'lamba_timeout': 10,
+                    'lambda_functions': {
+                        'myauth': {
+                            'lambda_timeout': 20,
+                            'lambda_memory_size': 512,
+                        }
+                    }
+                }
+            }
+        }
+        config = Config(
+            'dev',
+            config_from_disk=disk_config,
+            user_provided_params={'chalice_app': sample_app_with_auth,
+                                  'project_dir': '.'}
+        )
+        deployer = LambdaDeployer(
+            self.aws_client, self.packager, None, self.osutils,
+            self.app_policy)
+        self.aws_client.lambda_function_exists.return_value = False
+        self.aws_client.create_function.side_effect = [
+            self.lambda_arn, 'arn:auth-function']
+        deployer.deploy(config, None, stage_name='dev')
+        self.aws_client.create_function.assert_called_with(
```

Comment:
I'm not following.  That's what's happening on lines 900/901 for the timeout and memory size.  Was there more you wanted asserted?